src/App.js

```diff
@@ -1,20 +1,84 @@
-import cardsData from './assets/data.json';
+import React from 'react';
+import {getArticles} from './assets/helpers/get-articles';
 import Card from './components/Card.js';
+import NewCard from './components/NewCard.js';
+import Popup from './components/Popup';
+import baseStyles from './assets/styles/base.module.css';
 import styles from './App.module.css';
 
 
-function App() {
-    return (
-        <div className={styles.app}>
-            <main className={styles.pageContainer}>
-                {cardsData.map(item => (
-                    <div key={item.title} className={styles.card}>
-                        <Card cardData={item}></Card>
-                    </div>
-                ))}
-            </main>
-        </div>
-    );
+export class App extends React.Component {
+    state = {
+        loading: true,
+        cards: [],
+        creatingCard: false
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.onCreateCardClick = this.onCreateCardClick.bind(this);
+        this.onCardCreation = this.onCardCreation.bind(this);
+    }
+
+    componentDidMount() {
+        getArticles().then(cards => {
+            this.setState({
+                loading: false,
+                cards: cards
+            });
+        })
+    }
+
+    onCreateCardClick() {
+        this.setState({
+            ...this.state,
+            creatingCard: true
+        });
+    }
+
+    onCardCreation(title, description) {
+        this.setState({
+            ...this.state,
+            cards: this.state.cards.concat({
+                articleId: this.state.cards.reduce(
```

Comment:
articleId: Math.max(...oldState.cards.map(it => it.articleId)) + 1

Reply:
Done